src/Backend/OnboardingInsuranceAPI/Areas/Insurance/AddInsurance.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OnboardingInsuranceAPI.ErrorHandling;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class AddInsurance: IHandler
+{
+    private readonly DataContext _context;
+    private readonly ILogger<AddInsurance> _logger;
+
+    public AddInsurance(DataContext context, ILogger<AddInsurance> logger)
+    {
+        _context = context;
+        _logger = logger;   
+    }
+
+    public async Task Handle(InsuranceData item)
+    {
+        if (string.IsNullOrEmpty(item.Pid))
+        {
+            _logger.LogWarning("Error, received pid is empty!");
+            throw new ApiException(ErrorCode.InvalidQueryParameters);
+        }
+
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Pid == item.Pid);
+
+        if (user is null)
+        {
+            _logger.LogWarning("Error reading user, user is null!");
+            throw new ApiException(ErrorCode.NotFound);
+        }
+
+        user.CarInsurance = item.CarInsurance;
+        _context.Update(user);
```

Comment:
Also another thing you are just adding/overwriting the insurance. But what is user has more insurances? Also no calculation takes place so that user can see how much will the insurance cost.

Reply:
You are only reading and writing what frontend sends you. That is not usually how it goes.

Reply:
Hmmm, my app works with 1 insurance per user only, I have to think how to change frontend to extend to this possibility. About contribution its present as YearlyContribution but its calulating on frontend , I only did this "const contribution = yearProd.value * 2.33" so I can move to backend this calculation )